crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -381,61 +709,68 @@ pub async fn sys_validate_record(
     }
 }
 
-async fn sys_validate_record_inner(
-    record: &Record,
-    cascade: &impl Cascade,
-) -> SysValidationResult<()> {
+async fn sys_validate_record_inner<C>(record: &Record, cascade: Arc<C>) -> SysValidationResult<()>
+where
+    C: Cascade + Send + Sync,
+{
     let signature = record.signature();
     let action = record.action();
     let maybe_entry = record.entry().as_option();
     counterfeit_check(signature, action).await?;
 
-    async fn validate(
+    async fn validate<C>(
         action: &Action,
         maybe_entry: Option<&Entry>,
-        cascade: &impl Cascade,
-    ) -> SysValidationResult<()> {
-        let incoming_dht_ops_sender: Option<&IncomingDhtOpSender> = None;
-        store_record(action, cascade).await?;
+        cascade: Arc<C>,
+    ) -> SysValidationResult<()>
+    where
+        C: Cascade + Send + Sync,
+    {
+        let validation_dependencies = Arc::new(Mutex::new(ValidationDependencies::new()));
+        fetch_previous_actions(
```

Comment:
Note that the fetch here is free to go and access the network. The function doesn't care so it's up to the caller where they want this to go and look. This does mean there's a significant difference in the flow between sys validation the workflow and inline sys validation.